lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -708,6 +709,11 @@ def commit(request, options)
         # Set message to 'Failure' instead of 'Successful transaction' in that case.
         message = (!success && response_code == :r100) ? "Failure" : @@response_codes[response_code] rescue response[:message]
 
+        # if a soupFault is thrown there wont be a reasonCode
```

Comment:
is this soapFault ?

Reply:
technically its not a throw, its a high level security fault which is reflected in the structure of the xml i.e. '<soapFault>.... blah  </soapFault>'

Reply:
Fixed -> soapFault